src/devices/machine/acorn_vidc.cpp

```diff
@@ -0,0 +1,518 @@
+	//m_composite_sync = BIT(data, 7);
+	//m_test_mode = (data & 0xc100) != 0xc100;
+	
+	//todo: vga/svga modes sets 0x1000?
+	screen_vblank_line_update();
+	screen_dynamic_res_change();
+}
+
+WRITE32_MEMBER( acorn_vidc10_device::crtc_w )
+{
+	switch(offset)
+	{
+		case CRTC_HCR:  m_crtc_regs[CRTC_HCR] =  ((data >> 14)<<1)+2;    	break;
+//		case CRTC_HSWR: m_crtc_regs[CRTC_HSWR] = (data >> 14)+1;   			break;
+		case CRTC_HBSR: m_crtc_regs[CRTC_HBSR] = ((data >> 14)<<1)+1;    	break;
+		case CRTC_HDSR: m_crtc_regs[CRTC_HDSR] = (data >> 14);   			break;
+		case CRTC_HDER: m_crtc_regs[CRTC_HDER] = (data >> 14);   			break;
+		case CRTC_HBER: m_crtc_regs[CRTC_HBER] = ((data >> 14)<<1)+1;    	break;
+		case CRTC_HCSR: m_crtc_regs[CRTC_HCSR] = ((data >> 13) & 0x7ff) + 6; return;
+//      case CRTC_HIR: // ...
+
+		case CRTC_VCR:  m_crtc_regs[CRTC_VCR] = (data >> 14)+1; 			break;
+		case CRTC_VSWR: m_crtc_regs[CRTC_VSWR] = (data >> 14)+1;            break;
+		case CRTC_VBSR: 
+			m_crtc_regs[CRTC_VBSR] = (data >> 14)+1;
+			break;
+		case CRTC_VDSR: 
+			m_crtc_regs[CRTC_VDSR] = (data >> 14)+1;
+			break;
+		case CRTC_VDER:
+			m_crtc_regs[CRTC_VDER] = (data >> 14)+1;
+			screen_vblank_line_update();
+			break;
+		case CRTC_VBER: 
+			m_crtc_regs[CRTC_VBER] = (data >> 14)+1;
+			break;
+		case CRTC_VCSR: m_crtc_regs[CRTC_VCSR] = ((data >> 14) & 0x3ff) + 1; return;
+		case CRTC_VCER: m_crtc_regs[CRTC_VCER] = ((data >> 14) & 0x3ff) + 1; return;
+	}
+	
+	screen_dynamic_res_change();
+}
+
+inline void acorn_vidc10_device::refresh_stereo_image(uint8_t channel)
+{
+	/*
+		-111 full right
+		-110 83% right, 17% left
+		-101 67% right, 33% left
+		-100 center
+		-011 67% left, 33% right
+		-010 83% left, 17% right
+		-001 full left
+		-000 "undefined" TODO: verify what it actually means
+	*/
+	const float left_gain[8] = { 1.0, 2.0, 1.66, 1.34, 1.0, 0.66, 0.34, 0.0 };
+	const float right_gain[8] = { 1.0, 0.0, 0.34, 0.66, 1.0, 1.34, 1.66, 2.0 };
+	
+	m_lspeaker->set_input_gain(channel,left_gain[m_stereo_image[channel]]*m_sound_input_gain);
+	m_rspeaker->set_input_gain(channel,right_gain[m_stereo_image[channel]]*m_sound_input_gain);
+	//printf("%d %f %f\n",channel,m_lspeaker->input_gain(channel),m_rspeaker->input_gain(channel));
+}
+
+
+WRITE32_MEMBER( acorn_vidc10_device::stereo_image_w )
+{
+	uint8_t channel = (offset + 7) & 0x7;
+	m_stereo_image[channel] = data & 0x7;
+	refresh_stereo_image(channel);
+}
+
+WRITE32_MEMBER( acorn_vidc10_device::sound_frequency_w )
+{
+	m_sound_frequency_test_bit = BIT(data, 8);
+	m_sound_frequency_latch = data & 0xff;
+	if (m_sound_mode == true)
+		refresh_sound_frequency();
+}
+
+//**************************************************************************
+//  MEMC comms
+//**************************************************************************
+
+void acorn_vidc10_device::write_dac(uint8_t channel, uint8_t data)
+{ 
+	uint8_t ulaw_comp;
+	int16_t res;
+	const int16_t mulawTable[256] =
```

Comment:
Maybe better to make this a `static constexpr` to make it use BSS initialisation?

Reply:
Table should go away in any case: it's wrong and actually documented by page 48 of the VIDC20.